### PR TITLE
Names of RS and Pod generated by Deployments use encoded POD-TEMPLATE-HASH

### PR DESCRIPTION
**Pull request by islinwb**
closes https://github.com/kubernetes/website/issues/7577  (@zacharysarah changed "fixes" to "closes")

Use encoded POD-TEMPLATE-HASH instead of POD-TEMPLATE-HASH ( https://github.com/kubernetes/kubernetes/pull/51538).
I updated the naming part and examples. There are still some docs (such as [Application Introspection and Debugging](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-application-introspection/) use POD-TEMPLATE-HASH as part of the RS and Pod name. Some of the examples needs multi-node cluster which I currently don' have. And I think it doesn't matter that much for some docs.

### CODE REVIEW

**Comment by bradtopol** [comment]
Not sure what happened here. Going to try and rerun travis-ci
/retest

**Comment by zacharysarah** [comment]
@islinwb 👋 It's passing now!

File: docs/concepts/workloads/controllers/deployment.md [hunk 1]
```diff
@@ -346,20 +347,20 @@ nginx-deployment-2035384211) and new replicas (nginx-deployment-3066724191) are
...
-nginx-deployment-1564180365   2         2         0       25s
-nginx-deployment-2035384211   0         0         0       36s
-nginx-deployment-3066724191   2         2         2       6s
+nginx-deployment-5964dfd755   3         3         3       1m
+nginx-deployment-5d5cfdbd5f   1         1         0       20s
+nginx-deployment-6c54bd5869   0         0         0       3m
 ```
 
-Looking at the Pods created, you will see that the 2 Pods created by new ReplicaSet are stuck in an image pull loop.
+Looking at the Pods created, you will see that the 1 Pods created by new ReplicaSet are stuck in an image pull loop.
```
Comment (bradtopol) [inline docs/concepts/workloads/controllers/deployment.md]: Some grammatical errors. Should be "you will see that the Pod created by the new ReplicaSet is stuck in an image pull loop."

File: docs/concepts/workloads/controllers/deployment.md [hunk 2]
```diff
@@ -122,18 +122,18 @@ To see the ReplicaSet (`rs`) created by the deployment, run `kubectl get rs`:
 
 ```shell
 NAME                          DESIRED   CURRENT   READY   AGE
-nginx-deployment-2035384211   3         3         3       18s
+nginx-deployment-6c54bd5869   3         3         3       15s
 ```
 
-Notice that the name of the ReplicaSet is always formatted as `[DEPLOYMENT-NAME]-[POD-TEMPLATE-HASH-VALUE]`. The hash value is automatically generated when the Deployment is created.
+Notice that the name of the ReplicaSet is always formatted as `[DEPLOYMENT-NAME]-[ENCODED POD-TEMPLATE-HASH-VALUE]` (the format is `[DEPLOYMENT-NAME]-[POD-TEMPLATE-HASH-VALUE]` for Kubernetes v1.7 and before). The hash value is automatically generated when the Deployment is created.
```
Comment (bradtopol) [inline docs/concepts/workloads/controllers/deployment.md]: How this is rendering in the docs is actually hard to read.  And the information regarding v1.7 and before would best be documented using our callout style guidelines.  Please see https://kubernetes.io/docs/home/contribute/style-guide/#callout-formatting  for how to do this.

**Comment by bradtopol** [comment]
Glad to see this is passing Travis-CI.  Overall this is very good and while it looks like lots of changes many are just updating what outputted from the commands being run.  I did find a grammatical issue and style guideline issue that need to be addressed. Please see my inline comments.

File: docs/concepts/workloads/controllers/deployment.md (hunk 1 above)
Comment (islinwb) [reply docs/concepts/workloads/controllers/deployment.md]: @bradtopol Thanks for your review. I forgot to mention that in my testings there is only 1 pod which stucked in an image pull loop (not 2 pods described before).

**Comment by zacharysarah** [comment]
@bradtopol Bump

**Comment by zacharysarah** [comment]
@islinwb 👋 We've migrated the site's framework to Hugo. Please re-open this PR with changes to files in their post-migration location (`/content/en/docs/...`).

**Pull request closed without merging**